### PR TITLE
fix(3742): pagerduty test alerts

### LINKED ISSUE #3742: Notebooks(Alerts): PagerDuty test alert not being received
## About the bug

**Steps to reproduce:**
List the minimal actions needed to reproduce the behavior.

1. Create a PagerDuty account with an Events API v2 integration.  This will supply the integration/routing key
2. Go to Influxdbv2 UI (note that during this test the feature flag 'notebooksNewEndpoints' was set)
3. Add test data to a bucket
4. Create an Alert Notebook
5. Use the test data to define the query
6. In the alert section choose the PagerDuty endpoint
7. Provide the Routing Key from step 1 
8. Click the button 'Test Alert'

**Expected behavior:**

Expected to see a new incident appear in the PagerDuty console within at most a minute.  CURL test of the endpoint generated a new incident immediately. 

**Actual behavior:**

No alert message was received. 

**Visual Proof:**

TestNotificationPagerDuty01

## About your environment

**Environment info:**

Tested in K8S-IDPE Remocal. 
Latest UI commit
```
commit a918e664f19a21d33b21fe2c19c7cac8aa01c573 (upstream/master, origin/master, origin/HEAD, master)
Author: Palakp41 <palak@influxdata.com>
Date:   Tue Jan 25 17:16:27 2022 -0800
```
Lates K8S-IDPE commit
```
commit 6f99dc3e10c1365108336612dd032c746f1ddb9e (HEAD -> master, origin/master, origin/alchemy/automatic, origin/HEAD)
Author: Hercules Mango Churchill <alchemy@influxdata.com>
Date:   Mon Jan 31 08:51:32 2022 +0000

```

**Related Issue**
#3706

**Config:**

**Logs:**

### PR DESCRIPTION
**Pull request by wiedld**
Closes #3742

### Done:
* make pagerduty test alerts work.
* also update syntax for the pagerduty alerts exported to tasks.

### Visual proof:

https://user-images.githubusercontent.com/10232835/153685092-3f3f8134-6b8b-4597-8fd3-a372c4f48174.mov

### CODE REVIEW

**Review by asalem1 — APPROVED** [review]

File: src/flows/pipes/Notification/endpoints/PagerDuty/index.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export default register => {
 		messageFn: messageFn,
     crit: trigger,
 	)
-	|> monitor["notify"](data: notification, endpoint: pagerduty["endpoint"]()(mapFn: (r) => ({
+	|> monitor["notify"](data: notification, endpoint: pagerduty.endpoint()(mapFn: (r) => ({ r with
```
Comment (wiedld) [inline src/flows/pipes/Notification/endpoints/PagerDuty/index.ts:32]: not sure how I feel about the `r with` object extension.
It matches the docs...but it does potentially push client data over the wire.

File: src/flows/pipes/Notification/endpoints/PagerDuty/index.ts [hunk 2]
```diff
@@ -41,20 +41,24 @@ export default register => {
...
       eventAction: "trigger",
       source: "Notebook Generated Test Notification",
+      component: "example-component",
       summary: "${TEST_NOTIFICATION}",
-      timestamp: time(v: r["_source_timestamp"])
-}))
-  array.from(rows: [{value: 0}])
-	|> yield(name: "ignore")
-`,
+      timestamp: "${new Date().toISOString()}",
```
[the remark this answers is not in the document]
Comment (wiedld) [reply src/flows/pipes/Notification/endpoints/PagerDuty/index.ts:57]: nope. we don't (other than making it ISO.) resolving this.

File: src/flows/pipes/Notification/endpoints/PagerDuty/index.ts (hunk 1 above)
Comment (wiedld) [reply src/flows/pipes/Notification/endpoints/PagerDuty/index.ts:32]: confirmed. This is the intent. The alert templating system is then responsible for deciding.

**Comment by AzuladoToujours** [comment]
Hello @wiedld , I've been testing the alerts from Notebooks with the integration of PagerDuty. I'm currently using version 2.1.1 without success. Is this fix currently deployed in the version [2.2](https://github.com/influxdata/influxdata-docker/blob/e7469c2eae8f6a3dd50a2fbdea64aacde372cc74/influxdb/2.2/Dockerfile)?  Thanks!

**Comment by wiedld** [comment]
@AzuladoToujours yes this fix is deployed.
But note that we haven't shipped affiliated tickets based on form validation (and a few other things).

Do you mind describing in more detail, your exact failure? What steps are you taking leading up to the failure?
(This way I can reproduce & fix.) Thanks!

**Comment by AzuladoToujours** [comment]
@wiedld  I'm doing the same steps as the fix and the [issue](https://github.com/influxdata/ui/issues/3742) itself.

I'll upgrade to version 2.2 and keep you posted if the issue persists. Thanks! :)

**Pull request merged**